scripts/cleanup_analysis_entries.py

```diff
@@ -0,0 +1,378 @@
+from collections import defaultdict
+#from itertools import chain
+import logging
+import json
+import click
+from google.cloud import storage
+from sample_metadata.apis import ProjectApi, AnalysisApi
+from sample_metadata.model.analysis_query_model import AnalysisQueryModel
+from sample_metadata.model.analysis_type import AnalysisType
+
+
+# Global vars
+EXTENSIONS = ['.fastq.gz', '.fastq', '.bam', '.cram', '.fq', 'fq.gz']
+BUCKET_TYPES = [
+    'main',
+    'test',
+    'archive',
+    ' release',
+    'upload',
+    'tmp',
+    'analysis',
+    'web',
+]
+
+logger = logging.getLogger(__file__)
+logger.setLevel(level=logging.INFO)
+
+aapi = AnalysisApi()
+client = storage.Client()
+
+
+def get_datasets() -> list[dict]:
+    """API call to get project list"""
+    return ProjectApi().get_seqr_projects()
+
+
+def get_analyses_for_datasets(datasets, analysis_type) -> list[dict]:
+    """API call to get all analysis objects in datasets list"""
+    return aapi.query_analyses(
+        AnalysisQueryModel(
+            projects=datasets,
+            type=AnalysisType(analysis_type),
+            # status=AnalysisStatus('completed'),
+        )
+    )
+
+
+def get_path_components_from_path(path):
+    """
+    Return the {bucket_name}, {dataset}, and {subdir} for GS only paths
+
+    >>> get_bucket_name_path_from_path('gs://cpg-dataset-main/subfolder/subfolder2/my.cram')
+    {bucket_name:'cpg-dataset-main', dataset:'dataset', subdir:'subfolder/subfolder2'}
+
+    >>> get_bucket_name_path_from_path('gs://cpg-dataset-test-upload/my.cram')
+    {bucket_name:'cpg-dataset-test-upload', dataset:'dataset', subdir:''}
+    """
+    if not path.startswith('gs://'):
+        logging.error(f'Analysis path {path} not bucket storage.')
+        return
+
+    short_path = path.removeprefix('gs://').split('/', maxsplit=1)
+    bucket_name = short_path[0]
+
+    dataset_with_bucket_type = bucket_name.removeprefix('cpg-')
+
+    split_bucket_name = dataset_with_bucket_type.split('-')
+    for bucket_component in split_bucket_name:
+        if bucket_component in BUCKET_TYPES:
+            bt_index = split_bucket_name.index(bucket_component)
+
+    # bucket_type = '-'.join(split_bucket_name[bt_index:])
+    dataset = '-'.join(split_bucket_name[:bt_index])
+
+    subdir, _ = short_path[1].rsplit('/', 1)
```

Comment:
If you were feeling brave this would be an ideal regex situation...

```
import re

pattern = re.compile(
    r'gs://cpg-(?P<bucket>[a-zA-Z0-9\-]+)(?=-main|-test|-release|-archive)[a-zA-Z0-9\-]+/(?P<suffix>.+)/(?P<file>.*)'
)

test = 'gs://cpg-acute-care-main-web/reanalysis/folder/subfolder/aip_index.html'
mat = pattern.match(test)
print(mat.groups())

> ('acute-care', 'reanalysis/folder/subfolder', 'aip_index.html')
```

link to a RegEx101 sandbox: https://regex101.com/r/DQzojH/4

Reply:
I like the regex, I wouldn't match on specific bucket categories, just match anything after `main/test` as category if you wanted to return that.

Reply:
I was struggling to find a bulletproof way to make sure that the dataset name was separated from the bucket suffix without a lookahead match, given that the datasets also have an arbitrary number of hyphens 😓

Reply:
@MattWellie thanks heaps for the base to work from. This is what I came up with [regex101](https://regex101.com/r/94WYPd/1), it puts the bucket extension after main/test into its own group and then the code will handle if that group is none.

Reply:
I've tweaked that a bit further [here](https://regex101.com/r/94WYPd/2), but it's not clear to me from current usage of that method whether it's important to capture the extension separately from the rest of the bucket name (or if that's even valid to do, given that `cpg-dataset-main` and `cpg-dataset-main-web` are different `buckets`).

I've made a tweak on the original one [here](https://regex101.com/r/DQzojH/6) which separately captures the dataset, full bucket, suffix, and file (if required).